### ISSUE TITLE
Python 3.6: false positive W605 invalid escape sequence (pycodestyle 2.4.0)

### ISSUE THREAD

**Issue by kingbuzzman** [PyCQA/pycodestyle#814]
Similar issue to https://github.com/PyCQA/pycodestyle/issues/755

```shell
$ echo "re.compile('\{([^\}]+)\}')" | pycodestyle /dev/stdin
/dev/stdin:1:2: W605 invalid escape sequence '\{'
/dev/stdin:1:7: W605 invalid escape sequence '\}'
/dev/stdin:1:12: W605 invalid escape sequence '\}'
```

Working proof of concept:

```shell
$ echo "{hello} {world}" | python -c "import re, sys; r = re.compile('\{([^\}]+)\}'); print(r.findall(sys.stdin.read()))"
['hello', 'world']
```

**Comment by sigmavirus24** [comment]
In the future, please search **closed and** open issues before creating new ones that are duplicates. Also, you should be using raw strings

**Comment by elafontaine** [comment]
For people in the future landing here, see attached commit for the solution^
